streams/src/test/java/org/apache/kafka/streams/integration/SuppressionIntegrationTest.java

```diff
@@ -75,6 +86,74 @@
     private static final LongDeserializer LONG_DESERIALIZER = new LongDeserializer();
     private static final int COMMIT_INTERVAL = 100;
     private static final int SCALE_FACTOR = COMMIT_INTERVAL * 2;
+    private static final long TIMEOUT_MS = 30_000L;
+
+    @Ignore
+    @Test
+    public void shouldSuppressIntermediateEventsWithEmitAfter() throws InterruptedException {
+        final String testId = "-shouldSuppressIntermediateEventsWithEmitAfter";
+        final String appId = getClass().getSimpleName().toLowerCase(Locale.getDefault()) + testId;
+        final String input = "input" + testId;
+        final String outputSuppressed = "output-suppressed" + testId;
+        final String outputRaw = "output-raw" + testId;
+
+        cleanStateBeforeTest(input, outputSuppressed, outputRaw);
+
+        final StreamsBuilder builder = new StreamsBuilder();
+        final KTable<String, Long> valueCounts = builder
+            .table(
+                input,
+                Consumed.with(STRING_SERDE, STRING_SERDE),
+                Materialized.<String, String, KeyValueStore<Bytes, byte[]>>with(STRING_SERDE, STRING_SERDE)
+                    .withCachingDisabled()
+                    .withLoggingDisabled()
+            )
+            .groupBy((k, v) -> new KeyValue<>(v, k), Serialized.with(STRING_SERDE, STRING_SERDE))
+            .count(Materialized.<String, Long, KeyValueStore<Bytes, byte[]>>as("counts").withCachingDisabled());
+
+        valueCounts
+            .suppress(untilTimeLimit(ofMillis(scaledTime(2L)), unbounded()))
+            .toStream()
+            .to(outputSuppressed, Produced.with(STRING_SERDE, Serdes.Long()));
+
+        valueCounts
+            .toStream()
+            .filterNot((k, v) -> k.equals("tick"))
+            .to(outputRaw, Produced.with(STRING_SERDE, Serdes.Long()));
+
+        final KafkaStreams driver = getCleanStartedStreams(appId, builder);
+
+        try {
+            produceSynchronously(
+                input,
+                asList(
+                    new KeyValueTimestamp<>("k1", "v1", scaledTime(0L)),
+                    new KeyValueTimestamp<>("k1", "v2", scaledTime(1L)),
+                    new KeyValueTimestamp<>("k2", "v1", scaledTime(2L)),
+                    new KeyValueTimestamp<>("tick", "tick", scaledTime(5L))
+                )
+            );
+            verifyOutput(
+                outputRaw,
+                asList(
+                    new KeyValueTimestamp<>("v1", 1L, scaledTime(0L)),
+                    new KeyValueTimestamp<>("v1", 0L, scaledTime(1L)),
+                    new KeyValueTimestamp<>("v2", 1L, scaledTime(1L)),
+                    new KeyValueTimestamp<>("v1", 1L, scaledTime(2L))
+                )
+            );
+            verifyOutput(
+                outputSuppressed,
+                asList(
+                    new KeyValueTimestamp<>("v2", 1L, scaledTime(1L)),
```

Comment:
Why the supressed result will not have `tick` without filtering?

Reply:
Only because it isn't time to emit it yet. The `tick` is the newest event, at time 5, so it's still stream-time 5. Based on the suppression config, we won't emit it until after time 7.

The only reason I filtered it out of the "raw" output is that I felt it made the tests more confusing to read. Do you think I should also filter it from the suppressed output (even though it doesn't matter), just to avoid distracting readers with the question you raised?

Reply:
The reasoning makes sense.

I think not explicitly filtering `tick` and expect in the output raw is actually fine, and to me it is better exposing this behavior to make sure the raw data did include the `tick` value while the suppressed one does not.

Reply:
Ok, it's easy enough to put it back in.